src/network/serverpackethandler.cpp

```diff
@@ -1101,8 +1101,15 @@ void Server::handleCommand_Damage(NetworkPacket* pkt)
 				<< (int)damage << " hp at " << PP(player->getPosition() / BS)
 				<< std::endl;
 
-		playersao->setHP(playersao->getHP() - damage);
-		SendPlayerHPOrDie(playersao);
+		u8 oldhp = playersao->getHP();
+		playersao->setHP(oldhp - damage);
+
+		// Make sure that the player only dies once.
+		if (oldhp - playersao->getHP() > 0 && player->isDead()) {
```

Comment:
Is it just me or would this be like
`if (oldhp - (oldhp - damage) > 0 && player->hp == 0) {` that would be
`if (-damage > 0 && player->hp == 0) {`?